pkg/controller/jobs/pod/pod_controller.go

```diff
@@ -701,7 +700,7 @@ func (p *Pod) ReclaimablePods() ([]kueue.ReclaimablePod, error) {
 
 func IsPodOwnerManagedByKueue(p *Pod) bool {
 	if owner := metav1.GetControllerOf(&p.pod); owner != nil {
-		return jobframework.IsOwnerManagedByKueue(owner) || (owner.Kind == "RayCluster" && strings.HasPrefix(owner.APIVersion, "ray.io/v1alpha1"))
```

Comment:
I don't have context on why the `owner.Kind == "RayCluster"` check is here, but why are we removing it?

My guess is that this is here because owner reference of pods are always RayCluster, regardless if it was a RayCluster created from a RayJob. FYI @vicentefb we'll need to look at this for RayCluster support

Reply:
Actually, it's not removed. The check is moved to the `IsManagingObjectsOwner` callback of the RayJob integration in 8f289ae757ee0cd7c5b593415069e4a9666d127b, so the existing behaviour should not change there.

Reply:
Thanks!

Reply:
> Other than that, IsOwnerManagedByKueue is used to find the parent object of a Job that is also managed by Kueue (like JobSet).

That PR 1) moves the existing Ray related ad-hoc logic in the Pod controller to the RayJob controller by leveraging the `IsOwnerManagedByKueue` callback and 2) adds the `v1` group.

> IIUC, RayCluster was changed to use Job?

RayJob (not RayCluster) now uses Job to submit the ray job to the ray cluster.

> Is that part of the v1 update?

No, I haven't made that change, as it's an existing bug that's probably better to be fixed as part of #1434.

> In that case, a Job could be owned by a RayCluster. However, we don't kueue RayClusters, we queue RayJobs, so this logic might be broken

Right, I think so too, and propose we investigate it as part of #1434.